galasa-managers-parent/galasa-managers-core-parent/dev.galasa.core.manager/src/main/java/dev/galasa/core/manager/internal/properties/ResourceStringPattern.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright contributors to the Galasa project 
+ */
+package dev.galasa.core.manager.internal.properties;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import dev.galasa.core.manager.CoreManagerException;
+import dev.galasa.core.manager.internal.CorePropertiesSingleton;
+import dev.galasa.framework.spi.ConfigurationPropertyStoreException;
+import dev.galasa.framework.spi.cps.CpsProperties;
+
+
+/**
+ * Resource String Pattern
```

Comment:
Maybe worth expanding on the rps string build syntax. Just a basic guide for `{}` is how to define a char, what single values mean, and how to express complex char selection. Cant you do things like {a-zA-C2-5}? Which im not sure people would know, or where in the framework to find out how it works